src/main/java/org/icatproject/core/entity/FieldSet.java

```diff
@@ -0,0 +1,27 @@
+package org.icatproject.core.entity;
+
+import javax.xml.bind.annotation.XmlRootElement;
+import javax.xml.bind.annotation.XmlElement;
+import java.io.Serializable;
+
+@XmlRootElement
+public class FieldSet implements Serializable {
+    @XmlElement(name = "field")
```

Comment:
Minor: this will create a new entity type "fieldSet" having one single attribute "field". The value of that attribute is a list of the field values. In general, a fieldSet will have multiple field values. It might thus be more consistent to name that attribute "fields" (in plural).

Reply:
That makes sense - I originally named it `field` since that makes the XML look like:

```
<ns2:searchResponse xmlns:ns2="http://icatproject.org">
     <return xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="ns2:fieldSet">
         <field xmlns:xs="http://www.w3.org/2001/XMLSchema" xsi:type="xs:dateTime">2000-04-03T00:00:00+01:00</field>
         <field xmlns:xs="http://www.w3.org/2001/XMLSchema" xsi:type="xs:string">INVESTIGATION 1</field>
     </return>
     <return xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="ns2:fieldSet">
        <field xmlns:xs="http://www.w3.org/2001/XMLSchema" xsi:type="xs:dateTime">2000-06-04T00:00:00+01:00</field>
         <field xmlns:xs="http://www.w3.org/2001/XMLSchema" xsi:type="xs:string">INVESTIGATION 2</field>
     </return>
</ns2:searchResponse>
```

so the XML tag is singular for each `field`, but when interacting with it code side it makes more sense for it to be plural.

Reply:
I agree that there is no obvious solution for the best way to represent lists in the result. But note that with the `fields` in the plural, you are at least consistent with other replies from ICAT at the SOAP interface.

For instance, a search for an investigation including datasets yields something like:
```xml
    <ns2:searchResponse xmlns:ns2="http://icatproject.org">
      <return xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:type="ns2:investigation">
	<createId>simple/root</createId>
	<createTime>2021-02-19T11:29:37+01:00</createTime>
	<id>31</id>
	<!-- ... -->
	<datasets>
	  <createId>simple/root</createId>
	  <createTime>2021-02-19T11:29:37+01:00</createTime>
	  <id>81</id>
	  <!-- ... -->
	</datasets>
	<datasets>
	  <createId>simple/root</createId>
	  <createTime>2021-02-19T11:29:37+01:00</createTime>
	  <id>82</id>
	  <!-- ... -->
	</datasets>
	<!-- ... -->
      </return>
    </ns2:searchResponse>
```

And the result from a `getEntityInfo()` call looks like:
```xml
    <ns2:getEntityInfoResponse xmlns:ns2="http://icatproject.org">
      <return>
	<classComment>An investigation or experiment</classComment>
	<!-- ... -->
	<fields>
	  <name>id</name>
	  <notNullable>false</notNullable>
	  <relType>ATTRIBUTE</relType>
	  <type>Long</type>
	</fields>
	<fields>
	  <name>investigationInstruments</name>
	  <notNullable>false</notNullable>
	  <relType>MANY</relType>
	  <type>InvestigationInstrument</type>
	</fields>
	<!-- ... -->
      </return>
    </ns2:getEntityInfoResponse>
```